packages/satcheljs/lib/select.ts

```diff
@@ -0,0 +1,63 @@
+import {Reaction, Atom, IObservableValue, isObservableArray, isObservable} from 'mobx';
+
+export interface SelectorFunction {
+    [key: string]: () => any;
+}
+
+/**
+ * Decorator for action functions. Selects a subset from the state tree for the action.
+ */
+export default function select<StateShape>(selector: SelectorFunction) {
+    return function decorator<T extends Function>(target: T): T {
+        let _this = this;
+
+        let returnValue: any = function() {
+            let state = {};
+            let reaction = new Reaction('something', null);
```

Comment:
Oh haha! Yes, we should - but this is for internal tracking... supposed to just be unique. I'll add "Something" unique.